include/openthread/platform/settings.h

```diff
@@ -68,12 +68,11 @@ enum
     OT_SETTINGS_KEY_RESERVED             = 0x0006, ///< Reserved (previously auto-start).
     OT_SETTINGS_KEY_SLAAC_IID_SECRET_KEY = 0x0007, ///< SLAAC key to generate semantically opaque IID.
     OT_SETTINGS_KEY_DAD_INFO             = 0x0008, ///< Duplicate Address Detection (DAD) information.
-    OT_SETTINGS_KEY_OMR_PREFIX           = 0x0009, ///< Off-mesh routable (OMR) prefix.
+    OT_SETTINGS_KEY_BR_ULA_PREFIX        = 0x0009, ///< BR ULA prefix.
```

Comment:
I suggested replacing `OT_SETTINGS_KEY_OMR_PREFIX` with `OT_SETTINGS_KEY_BR_ULA_PREFIX` because it should work. 

But agree that we should follow the guidelines that were well established. 

We should add a new key for BR_ULA_PREFIX, and expire the old keys for OMR and NAT64.

Reply:
Right that reading OMR prefix as `BR_ULA` will not pass the validatation. But it's not strictly backward compatible.

Updated to reserve `0x0009` for legacy OMR prefix.  
And in `routing_manager.cpp`, `Delete` is called for legacy OMR prefix to clear the storage on device.